api/client.go

```diff
@@ -173,7 +180,23 @@ func DefaultConfig() *Config {
 		Backoff:      retryablehttp.LinearJitterBackoff,
 	}
 
+	if err := config.ReadEnvironment(); err != nil {
```

Comment:
I'm a bit nervous about moving where we call ReadEnvironment.  I haven't identified any obvious issues with it, but I haven't been able to rule out the possibility of it breaking something.  Can we instead have ReadEnvironment include the assignment to transport.Proxy?

Reply:
I also don't see any reason why moving it would matter, but it's easy enough to apply the override in `ReadEnvironment`. I've updated to do that instead.